### PR TITLE
User authentication specs

### PR DESCRIPTION
**Pull request by prateeksan**
#13

### CODE REVIEW

**Comment by prateeksan** [comment]
@mrdomino Ill write feature specs for session management when I start doing session management. Im not sure how to go about it yet.

File: spec/features/user_registration_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# /spec/features/user_registration_spec.rb
+require 'rails_helper'
+
+RSpec.feature 'New User Registration' do
```
Comment (mrdomino) [inline spec/features/user_registration_spec.rb]: I like the "<entity> <does thing>" form for features. E.g. "Visitor registers" in this case. But it's just a matter of taste of course.

File: spec/features/user_registration_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  
+  scenario 'A new user registers with valid fields' do
```
Comment (mrdomino) [inline spec/features/user_registration_spec.rb]: I think this can just be "with valid fields"

File: spec/features/user_registration_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+      click_button "Sign Up"
+      expect(page).to have_text("Welcome testuser!")
+  end
+
+  scenario 'A new user registers with invalid fields' do
+      visit '/'
+      click_link "Sign Up"
+
+      fill_in "Username", :with => "testuser"
```
Comment (mrdomino) [inline spec/features/user_registration_spec.rb:30]: I'm sort of tempted to extract the fields into a given block at the top, like: `given(:user) { build(:user) }` (with a `spec/factories/user.rb` that provides default values for all the fields), and then filling all the fields in with `user.field`. Then it'd be clearer by looking at it what's different between the valid and invalid scenario. But it's optional. Maybe add a TODO.

File: spec/features/user_registration_spec.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      fill_in "Street Address", :with => "5 Street West"
+      fill_in "Biodata", :with => "All you need to know about me."
+
+      #TODO(soon): Add spec tests for Avatar upload
+
+      click_button "Sign Up"
+      expect(page).to have_text("The following errors occured:")
+  end
+
+end
```
Comment (mrdomino) [inline spec/features/user_registration_spec.rb:46]: super-trivial nitpick: no trailing newline

**Comment by mrdomino** [comment]
Ok to merge. Up to you how you want to integrate any of the feedback -- I'm okay with it going in as is.

**Pull request merged**